clients/src/test/java/org/apache/kafka/clients/InFlightRequestsTest.java

```diff
@@ -0,0 +1,188 @@
+        String nodeId = (nodeIdCounter++).toString();
+        int totalRequest = 10;
+        pushRequests(nodeId, totalRequest);
+        assertEquals(totalRequest, inFlightRequests.count(nodeId));
+    }
+
+    @Test
+    public void testCompleteNext() {
+        NetworkClient.InFlightRequest oldestRequest = requests.remove(0);
+        decrementCounter(oldestRequest.destination);
+        assertEquals(oldestRequest, inFlightRequests.completeNext(oldestRequest.destination));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testCompleteNextWithInvalidNode() {
+        inFlightRequests.completeNext("");
+    }
+
+
+    @Test
+    public void testLastSent() {
+        NetworkClient.InFlightRequest lastSentRequest = requests.get(requests.size() - 1);
+        assertEquals(lastSentRequest, inFlightRequests.lastSent(lastSentRequest.destination));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testLastSentWithInvalidNode() {
+        inFlightRequests.lastSent("");
+    }
+
+    @Test
+    public void testCompleteLastSent() {
+        NetworkClient.InFlightRequest lastSentRequest = requests.remove(requests.size() - 1);
+        decrementCounter(lastSentRequest.destination);
+        assertEquals(lastSentRequest, inFlightRequests.completeLastSent(lastSentRequest.destination));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testCompleteLastSentWithInvalidNode() {
+        inFlightRequests.completeLastSent("");
+    }
+
+    @Test
+    public void testCanSendMore() {
+    }
+
+    @Test
+    public void testCount() {
+        assertEquals(requests.size(), inFlightRequests.count());
+    }
+
+    @Test
+    public void testIsEmpty() {
+        assertEquals(inFlightRequests.count() == 0, inFlightRequests.isEmpty());
+    }
+
+    @Test
+    public void testCountInSingleNode() {
+        String nodeId = (nodeIdCounter++).toString();
+        int totalRequest = 10;
+        pushRequests(nodeId, totalRequest);
+        assertEquals(totalRequest, inFlightRequests.count(nodeId));
+    }
+
+    @Test
+    public void testIsEmptyInSingleNode()  {
+        String nodeId = (nodeIdCounter++).toString();
+        assertTrue(inFlightRequests.isEmpty(nodeId));
+
+        nodeId = (nodeIdCounter++).toString();
+        int totalRequest = 10;
+        pushRequests(nodeId, totalRequest);
+
+        assertEquals(totalRequest, inFlightRequests.count(nodeId));
+        inFlightRequests.clearAll(nodeId);
+        assertTrue(inFlightRequests.isEmpty(nodeId));
+    }
+
+    @Test
+    public void testClearAll() {
+        String nodeId = (nodeIdCounter++).toString();
+        int totalRequest = 10;
+        pushRequests(nodeId, totalRequest);
+
+        assertEquals(totalRequest, inFlightRequests.count(nodeId));
+        inFlightRequests.clearAll(nodeId);
+        assertEquals(0, inFlightRequests.count(nodeId));
+    }
+
+    @Test
+    public void testGetNodesWithTimedOutRequests() {
```

Comment:
Are you planning to add this? If you add `time` as an instance variable set to `MockTime()`, you should be able to easily test after calling `time.sleep()`.